service/src/main/java/bio/terra/cbas/retry/RetryLoggingListener.java

```diff
@@ -0,0 +1,36 @@
+package bio.terra.cbas.retry;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.retry.RetryCallback;
+import org.springframework.retry.RetryContext;
+import org.springframework.retry.RetryListener;
+import org.springframework.stereotype.Component;
+
+@Component("retryLoggingListener")
+public class RetryLoggingListener implements RetryListener {
+
+  private final Logger logger = LoggerFactory.getLogger(getClass());
+
+  @Override
+  public <T, E extends Throwable> void close(
+      RetryContext context, RetryCallback<T, E> callback, Throwable throwable) {
+    if (context.getRetryCount() > 1) {
+      logger.debug("Retryable method closing ({}th retry)", context.getRetryCount());
```

Comment:
nit: these `({}th retry)` messages will look funny for retry number 1, 2, and 3! I wouldn't block approval over this, but you could do `(retry number {})` or `(retry #{})`